### PR TITLE
Fix deprecated parameters

### PR DESCRIPTION
**Pull request by Badiboy**
## Description
Fix deprecated parameters:
1. Fix descriptions
2. Fix bot.reply_to function
3. Fix ChatPermissions.de_json

### CODE REVIEW

File: telebot/__init__.py [hunk 1]
```diff
@@ -5255,7 +5249,18 @@ def reply_to(self, message: types.Message, text: str, **kwargs) -> types.Message
...
         """
-        return self.send_message(message.chat.id, text, reply_to_message_id=message.message_id, **kwargs)
+        if kwargs:
+            reply_parameters = kwargs.pop("reply_parameters", None)
+        else:
+            reply_parameters = None
+        if not reply_parameters:
+            reply_parameters = types.ReplyParameters(message.message_id)
+
+        if "allow_sending_without_reply" in kwargs:
```
Comment (coder2020official) [inline telebot/__init__.py:5259]: why don't we check if allow_sending_without_reply is already set?

File: telebot/async_telebot.py [hunk 2]
```diff
@@ -6237,7 +6232,18 @@ async def reply_to(self, message: types.Message, text: str, **kwargs) -> types.M
...
         """
-        return await self.send_message(message.chat.id, text, reply_to_message_id=message.message_id, **kwargs)
+        if kwargs:
+            reply_parameters = kwargs.pop("reply_parameters", None)
+        else:
+            reply_parameters = None
+        if not reply_parameters:
+            reply_parameters = types.ReplyParameters(message.message_id)
+
+        if "allow_sending_without_reply" in kwargs:
```
Comment (coder2020official) [inline telebot/async_telebot.py:6242]: same here. let's check for conflicting parameters

**Review by coder2020official — CHANGES_REQUESTED** [review]

File: telebot/__init__.py [hunk 3]
```diff
@@ -3547,48 +3547,43 @@ def restrict_chat_member(
...
-        :param can_pin_messages: Pass True, if the user is allowed to pin messages. Ignored in public supergroups
+        :param can_pin_messages: deprecated
         :type can_pin_messages: :obj:`bool`
 
-        :param use_independent_chat_permissions: Pass True if chat permissions are set independently. Otherwise,
-            the can_send_other_messages and can_add_web_page_previews permissions will imply the can_send_messages,
-            can_send_audios, can_send_documents, can_send_photos, can_send_videos, can_send_video_notes, and
-            can_send_voice_notes permissions; the can_send_polls permission will imply the can_send_messages permission.
-        :type use_independent_chat_permissions: :obj:`bool`
+        :param use_independent_chat_permissions: Optional	Pass True if chat permissions are set independently.
```
[the remark this answers is not in the document]
Comment (Badiboy) [reply telebot/__init__.py:3574]: Multiline. Tab spaces is the same as any other multiline descriptions.

[the remark this answers is not in the document]
Comment (coder2020official) [reply telebot/__init__.py:3574]: Not sure whether doc will work with that

[the remark this answers is not in the document]
Comment (coder2020official) [reply telebot/__init__.py:3574]: Why not just make Optional: Pass ...

[the remark this answers is not in the document]
Comment (Badiboy) [reply telebot/__init__.py:3574]: image

Come on. There are lot of descriptions like that.

[the remark this answers is not in the document]
Comment (coder2020official) [reply telebot/__init__.py:3574]: what do you mean? tab is being used on the **next** line

File: telebot/__init__.py (hunk 1 above)
Comment (Badiboy) [reply telebot/__init__.py:5259]: "Optional	Pass" - you mean here? I see, will fix.

File: telebot/__init__.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Badiboy) [reply telebot/__init__.py:3574]: "Optional       Pass" - you mean here? I see, will fix. I copied from TG :)

[the remark this answers is not in the document]
Comment (coder2020official) [reply telebot/__init__.py:3574]: lol yes

File: telebot/__init__.py (hunk 1 above)
Comment (Badiboy) [reply telebot/__init__.py]: +

File: telebot/async_telebot.py (hunk 2 above)
Comment (Badiboy) [reply telebot/async_telebot.py]: +

**Review by coder2020official — APPROVED** [review]

**Pull request merged**